pkg/pillar/agentlog/loglevel.go

```diff
@@ -39,16 +39,22 @@ func getLogLevelImpl(sub *pubsub.Subscription, agentName string,
 		log.Infof("GetLogLevel failed %s\n", err)
 		return "", false
 	}
-	gc := cast.CastGlobalConfig(m)
-	// Do we have an entry for this agent?
-	as, ok := gc.AgentSettings[agentName]
-	if ok && as.LogLevel != "" {
-		return as.LogLevel, true
-	}
-	// Do we have a default value?
-	if allowDefault && gc.DefaultLogLevel != "" {
-		return gc.DefaultLogLevel, true
+	gc := cast.CastConfigItemValueMap(m)
+
+	if _, ok1 := gc.AgentSettings[agentName]; ok1 {
```

Comment:
Why are you ignoring this? In the next sentence, you are doing the gc.AgentSettings[agentName] lookup again.. 2 lookups.. Instead:

if as, ok := gc.AgentSettings[agentName]; ok {
   if logSetting, ok := as["loglevel"]
   if ok {
         loglevel := logSetting.StringValue
         if loglevel != "" {
             ok = fmt.Errorf("Invalid logLevel (empty string) in agent setting for %s", agentName)
         } 
     }
     if ok != nil {
         // per-agent value not found.. Use Global agent setting
     }
  }
}

Somethign liek this is what we need.